ocaml/idl/datamodel.ml

```diff
@@ -3891,16 +3892,120 @@ let host_crashdump =
     ] 
     ()
 
-(* New Miami pool patching mechanism *)
+(* New Ely pool update mechanism *)
 
-let pool_patch_after_apply_guidance =
+let pool_update_after_apply_guidance =
   Enum ("after_apply_guidance", 
-        [ "restartHVM",  "This patch requires HVM guests to be restarted once applied.";
-          "restartPV",   "This patch requires PV guests to be restarted once applied.";
-		  "restartHost", "This patch requires the host to be restarted once applied."; 
-          "restartXAPI", "This patch requires XAPI to be restarted once applied.";
+        [ "restartHVM",  "This update requires HVM guests to be restarted once applied.";
+          "restartPV",   "This update requires PV guests to be restarted once applied.";
+		  "restartHost", "This update requires the host to be restarted once applied."; 
+          "restartXAPI", "This update requires XAPI to be restarted once applied.";
         ])
 
+let pool_update_introduce = call
+  ~name:"introduce"
+  ~doc:"Introduce update VDI"
+  ~in_oss_since:None
+  ~in_product_since:rel_ely
+  ~params:[ Ref _vdi, "vdi", "The VDI which contains update" ]
+  ~result:(Ref _pool_update, "the introduced pool update")
+  ~allowed_roles:_R_POOL_OP
+  ()
+
+let pool_update_precheck = call
+  ~name:"precheck"
+  ~doc:"Execute the precheck stage of the selected update on a host"
+  ~in_oss_since:None
+  ~in_product_since:rel_ely
+  ~params:[ Ref _pool_update, "self", "The update whose prechecks will be run"; Ref _host, "host", "The host to run the prechecks on" ]  
+  ~allowed_roles:_R_POOL_OP
+  ~forward_to:(HostExtension "Pool_update.precheck")
+  ()
+
+let pool_update_apply = call
+  ~name:"apply"
+  ~doc:"Apply the selected update to a host and return its output"
+  ~in_oss_since:None
+  ~in_product_since:rel_ely
+  ~params:[ Ref _pool_update, "self", "The update to apply"; Ref _host, "host", "The host to apply the update too" ]
+  ~allowed_roles:_R_POOL_OP
+  ~forward_to:(HostExtension "Pool_update.apply")
+  ()
+
+let pool_update_pool_apply = call
+  ~name:"pool_apply"
+  ~doc:"Apply the selected update to all hosts in the pool and return a map of host_ref -> update output"
```

Comment:
In my understanding, leave `~result` here means that we will restrict the return type is `unit`
refer to

1. https://github.com/xenserver/xen-api/blob/bd036f8ab02af8067b9f5681b1b58c7770b60d32/ocaml/idl/datamodel.ml#L373
1. https://github.com/xenserver/xen-api/blob/bd036f8ab02af8067b9f5681b1b58c7770b60d32/ocaml/idl/ocaml_backend/gen_server.ml#L96

Please correct me if any misunderstanding.